cloudinit/config/schema.py

```diff
@@ -387,6 +387,7 @@ def validate_cloudconfig_schema(
     strict_metaschema: bool = False,
     log_details: bool = True,
     log_deprecations: bool = False,
+    throw: bool = False,
```

Comment:
I guess we want to pass this arg to `validate_cloudconfig_metaschema`, right?

Additionally, to keep consistent the docstr, I would add documentation about this param.

Reply:
Thanks @aciba90 this was a vestigial addition when I thought about refactoring the exception handling inside `validate_cloudconfig_file` to avoid interleaving print statements for deprecation vs general exception handling. I opted not to refactor that in this branch, and instead perform some indenting of output as we enumerate user-data, vendor-data and vendor2-data instead.

The output now looks like the following:

## valid vendor-data and user-data no annotation
```
$ cloud-init schema --system --annotate
Found cloud-config data types: user-data, vendor-data

1. user-data at /var/lib/cloud/instances/a5f4a923-4f17-4611-b34c-f59b5da5c8cb/cloud-config.txt:
  Valid cloud-config: user-data

2. vendor-data at /var/lib/cloud/instances/a5f4a923-4f17-4611-b34c-f59b5da5c8cb/vendor-cloud-config.txt:
  Valid cloud-config: vendor-data

## valid user-data invalid vendor-data 
```
$ cloud-init schema --system
Found cloud-config data types: user-data, vendor-data

1. user-data at /var/lib/cloud/instances/a5f4a923-4f17-4611-b34c-f59b5da5c8cb/cloud-config.txt:
  Valid cloud-config: user-data

2. vendor-data at /var/lib/cloud/instances/a5f4a923-4f17-4611-b34c-f59b5da5c8cb/vendor-cloud-config.txt:
  Invalid cloud-config /var/lib/cloud/instances/a5f4a923-4f17-4611-b34c-f59b5da5c8cb/vendor-cloud-config.txt
  Error: Cloud config schema errors: users.1: {'lock_passwd': False, 'name': 'root', 'ssh_pwauth': True} is not valid under any of the given schemas

Error: Invalid cloud-config schema: vendor-data
```
## valid user-data, invalid vendor-data with --annotate
```
Found cloud-config data types: user-data, vendor-data

1. user-data at /var/lib/cloud/instances/a5f4a923-4f17-4611-b34c-f59b5da5c8cb/cloud-config.txt:
  Valid cloud-config: user-data

2. vendor-data at /var/lib/cloud/instances/a5f4a923-4f17-4611-b34c-f59b5da5c8cb/vendor-cloud-config.txt:
#cloud-config

# from 1 files
# part-001

---
users:
- default
-   lock_passwd: false		# E1
    name: root
    ssh_pwauth: true
...

# Errors: -------------
# E1: {'lock_passwd': False, 'name': 'root', 'ssh_pwauth': True} is not valid under any of the given schemas
```